api/src/shop/stripe_product_price.py

```diff
@@ -24,25 +24,28 @@
 
 
 def makeradmin_to_stripe_recurring(makeradmin_product: Product, price_type: PriceType) -> StripeRecurring | None:
+    subscription_category_id = get_subscription_category().id
     if price_type == PriceType.RECURRING or price_type == PriceType.BINDING_PERIOD:
+        if makeradmin_product.category.id != subscription_category_id:
+            raise ValueError(f"Unexpected price type {price_type} for non-subscription product {makeradmin_product.id}")
         if makeradmin_product.unit in makeradmin_unit_to_stripe_unit:
             interval = makeradmin_unit_to_stripe_unit[makeradmin_product.unit]
         else:
             raise ValueError(f"Unexpected unit {makeradmin_product.unit} in makeradmin product {makeradmin_product.id}")
         interval_count = makeradmin_product.smallest_multiple if price_type == PriceType.BINDING_PERIOD else 1
         return StripeRecurring(interval=interval, interval_count=interval_count)
     else:
+        if makeradmin_product.category.id == subscription_category_id:
+            raise ValueError(f"Unexpected price type {price_type} for subscription product {makeradmin_product.id}")
         return None
 
 
 def get_stripe_product_id(makeradmin_product: Product) -> str:
-    prefix = "debug" if debug_mode() else "prod"
-    return f"{prefix}_{makeradmin_product.id}"
```

Comment:
We could set in the metadata or description but this particular part with the id was removed by the other PR.
I've added checks at startup that checks the stripe key and debug mode so that prod key with debug mode will ask for conformation before continuing